src/langcheck/metrics/eval_clients/_litellm.py

```diff
@@ -0,0 +1,509 @@
+from __future__ import annotations
+
+import asyncio
+from typing import Any, Literal
+
+import instructor
+import torch
+from litellm import acompletion, aembedding, completion, embedding
+from litellm.types.utils import EmbeddingResponse
+from openai.types.chat import ChatCompletionMessageParam
+from pydantic import BaseModel
+
+from langcheck.utils.progress_bar import tqdm_wrapper
+
+from ..prompts._utils import get_template
+from ..scorer._base import BaseSimilarityScorer
+from ._base import EvalClient, TextResponseWithLogProbs
+from .extractor import Extractor
+
+
+class LLMEvalClient(EvalClient):
+    """EvalClient defined for OpenAI API."""
+
+    def __init__(
+        self,
+        model: str,
+        embedding_model: str | None = None,
+        *,
+        api_key: str | None = None,
+        api_base: str | None = None,
+        api_version: str | None = None,
+        vertex_location: str | None = None,
+        vertex_credentials: str | None = None,
+        aws_access_key_id: str | None = None,
+        aws_secret_access_key: str | None = None,
+        aws_region_name: str | None = None,
+        use_async: bool = False,
+        system_prompt: str | None = None,
+        extractor: Extractor | None = None,
+    ):
+        """
+        Initialize the LLM evaluation client.
```

Comment:
I think we should explain the arguments more in detail here! Especially `use_async`, `system_prompt` and `extractor` are the arguments that do not exist in `litellm` itself. The naming rules of `model` (e.g. you need to start the model names with `api_provider/` is a bit unclear as well.